routers/users.py

```
@@ -132,11 +132,26 @@ async def delete_account(current_user: dict = Depends(auth.get_current_user)):
+    for post in posts:
+        if post["id"] in user_commented_post_ids:
+            count = sum(1 for c in comments if c.get("user_id") == user_id and c.get("post_id") == post["id"])
+            post["comment_count"] = max(post.get("comment_count", 0) - count, 0)
+
     comments = [c for c in comments
                 if c.get("user_id") != user_id and c.get("post_id") not in user_post_ids]
     data.save_data(comments, "comments.json")
 
     likes = data.load_data("likes.json")
+
+    user_liked_post_ids = [l["post_id"] for l in likes
```

Ruff (0.14.11)

Ambiguous variable name: `l`

(E741)